### PR TITLE
feat: created custom help message

### PR DESCRIPTION
**Pull request by Souvikns**
**Description**

Add the reverted changes from #124

### CODE REVIEW

**Comment by Souvikns** [comment]
when I was merging that old branch to any other branch it was overwriting the commits that were ahead. So I created another branch from the upstream/main and then manually merged the changes from that merge. I think it should work now, After this, I will re fork again.

File: package.json
Comment (fmvilas) [inline package.json:89]: That's not going to work. See my comment in the previous PR: https://github.com/asyncapi/cli/pull/103#issuecomment-958718524

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: package.json
Comment (Souvikns) [reply package.json:89]: Hey, I checked everything, and I think that this PR is alright it is working as intended. Let me explain, so I guess there is something is weird with the previous branch. What was happening is that the upstream branch has moved forward and so did that branch, So if I pull the changes from the upstream branch it should add the commits that have been made since I created the initial branch. But this was not happening and the upstream branch was getting replaced with the branch that I created resulting in the deletion of the `studio` code. 

So in a attempt to fix this I update my Forked main branch with the `upstream/main` branch so now they are equal and when I pull the `new-help-string` branch then it is doing the same thing and replacing my master with that `new-help-string` branch. 

So this is a new branch from the `upsteam/master` and then I manually added the changes. I checked by merging this branch with the master and it was working as intended. I ran all the tests they are working have all the `studio` code and the help string is updated.

Comment (Souvikns) [reply package.json:89]: Oh, I get it, my bad 😄

Comment (fmvilas) [reply package.json]: No problem :)

**Review by fmvilas — APPROVED** [review]

File: src/help/command.ts
Comment (magicmatatjahu) [inline src/help/command.ts:13]:
```suggestion
const {
  underline,
  bold,
  dim,
} = chalk;
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
If I pass to console:

```bash
./bin/run help
```

then I see 

```text
display help for asyncapi

USAGE
  $ asyncapi help [COMMAND]

ARGUMENTS
  COMMAND  command to show help for

FLAGS
  --all  see all commands in CLI

All in one CLI for all AsyncAPI tools

VERSION
  @asyncapi/cli/0.8.1 darwin-x64 node-v14.16.1

USAGE
  $ asyncapi [COMMAND]

COMMANDS
  config    access configs
  help      display help for asyncapi
  new       creates a new asyncapi file
  validate  validate asyncapi file
```

and then if I pass 

```bash
./bin/run help new
```

I only see:

```
display help for asyncapi

USAGE
  $ asyncapi help [COMMAND]

ARGUMENTS
  COMMAND  command to show help for

FLAGS
  --all  see all commands in CLI
```

Should I see all options (and description) for `new` command? Or will it be added in the next PRs?

**Comment by Souvikns** [comment]
@boyney123 can you help me with this, I am getting this error in the action, it is running fine in my local setup. The error is originating from scripts. 

```

> @asyncapi/cli@0.10.0 build /home/runner/work/cli/cli
> node scripts/fetch-asyncapi-example.js && tsc

internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module 'request'
Require stack:
- /home/runner/work/cli/cli/scripts/fetch-asyncapi-example.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/home/runner/work/cli/cli/scripts/fetch-asyncapi-example.js:3:17)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/home/runner/work/cli/cli/scripts/fetch-asyncapi-example.js' ]
}
```

**Comment by boyney123** [comment]
Hey @Souvikns ,

Yeah, that's very odd! Just checked on my machine and working fine, 🤔 also the `request` module is in the `package.json` and lock file.....

Interesting warning we get on that log stack 

```
npm WARN Local package.json exists, but node_modules missing, did you mean to install?
```

Looks like it might be failing/missing to install the node_modules (run npm install?)?

**Comment by boyney123** [comment]
```
npm info it worked if it ends with ok
npm verb cli [
npm verb cli   '/opt/hostedtoolcache/node/14.18.1/x64/bin/node',
npm verb cli   '/opt/hostedtoolcache/node/14.18.1/x64/bin/npm',
npm verb cli   'install',
npm verb cli   '--loglevel',
npm verb cli   'verbose'
npm verb cli ]
npm info using npm@6.14.15
npm info using node@v14.18.1
npm verb npm-session 88392f2cfdf4e5e3
npm info lifecycle @asyncapi/cli@0.10.0~preinstall: @asyncapi/cli@0.10.0
npm timing stage:loadCurrentTree Completed in 18ms
npm timing stage:loadIdealTree:cloneCurrentTree Completed in 1ms
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
npm timing stage:loadIdealTree:loadShrinkwrap Completed in 1331ms
npm http fetch GET 200 registry.npmjs.org/fsevents 137ms
npm http fetch GET 200 registry.npmjs.org/fsevents/-/fsevents-2.3.2.tgz 47ms
npm timing stage:rollbackFailedOptional Completed in 0ms
npm timing stage:runTopLevelLifecycles Completed in 2040ms
npm verb stack Error: Unsupported platform for fsevents@2.3.2: wanted {"name":"fsevents","version":"2.3.2","engines":{"node":"^8.16.0 || ^10.6.0 || >=11.0.0"},"os":["darwin"],"dependencies":{},"optionalDependencies":{},"peerDependenciesMeta":{},"devDependencies":{"node-gyp":"^6.1.0"},"bundleDependencies":false,"peerDependencies":{},"deprecated":false,"_resolved":"registry.npmjs.org/fsevents/-/fsevents-2.3.2.tgz","_integrity":"sha512-xiqMQR4xAeHTuB9uWm+fFRcIOgKBMiOBP+eXiyT7jsgVCq1bkVygt00oASowB7EdtpOHaaPgKt812P9ab+DDKA==","_shasum":"8a526f78b8fdf4623b709e0b975c52c24c02fd1a","_shrinkwrap":null,"_id":"fsevents@2.3.2","_from":"fsevents@~2.3.2","_requested":{"type":"range","registry":true,"raw":"fsevents@~2.3.2","name":"fsevents","escapedName":"fsevents","rawSpec":"~2.3.2","saveSpec":null,"fetchSpec":"~2.3.2"},"_spec":"fsevents@~2.3.2","_where":"/home/runner/work/cli/cli/node_modules/chokidar"} (current: {"os":"linux","cpu":"x64"})
npm verb stack     at checkPlatform (/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/npm-install-checks/index.js:46:14)
npm verb stack     at thenWarnEngineIssues (/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/lib/install/validate-args.js:63:5)
npm verb stack     at /opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/iferr/iferr.js:3:64
npm verb stack     at checkEngine (/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/npm-install-checks/index.js:25:10)
npm verb stack     at module.exports.isInstallable (/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/lib/install/validate-args.js:60:3)
npm verb stack     at resolveWithNewModule (/opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/lib/install/deps.js:677:10)
npm verb stack     at /opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/lib/install/deps.js:600:7
npm verb stack     at /opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/iferr/iferr.js:3:64
npm verb stack     at /opt/hostedtoolcache/node/14.18.1/x64/lib/node_modules/npm/node_modules/call-limit/call-limit.js:28:43
npm verb stack     at processTicksAndRejections (internal/process/task_queues.js:77:11)
npm verb pkgid fsevents@2.3.2
npm verb cwd /home/runner/work/cli/cli
npm verb Linux 5.11.0-1021-azure
npm verb argv "/opt/hostedtoolcache/node/14.18.1/x64/bin/node" "/opt/hostedtoolcache/node/14.18.1/x64/bin/npm" "install" "--loglevel" "verbose"
npm verb node v14.18.1
npm verb npm  v6.14.15
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
npm verb exit [ 1, true ]
npm timing npm Completed in 4447ms

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-11-26T11_21_43_435Z-debug.log
Error: Process completed with exit code 1.
```

Interesting looks like `npm install` is not working correctly

**Comment by boyney123** [comment]
@Souvikns not sure if it's related to your PR, but looks like problems with `fsevents` which is only supported on mac OSX.

This windows test environment is failing. (More details here https://stackoverflow.com/questions/48898312/npm-error-unsupported-platform-for-fsevents-when-deploying-to-azure).

There is some suggestions in this solution that we could tell it to make `fsevents` an optional dep, which might get skipped on windows... but not sure, do you want to give it a try?

**Comment by Souvikns** [comment]
It is also running in my local environment, and I am using ubuntu. I am giving it a try with what you suggested.

File: src/help/command.ts
Comment (Souvikns) [reply src/help/command.ts]: I have moved the help command now it does not show in the root help message.

**Comment by Souvikns** [comment]
@boyney123 thanks it is working now 👍🏻

**Comment by Souvikns** [comment]
@derberg @fmvilas @magicmatatjahu waiting for your review.

**Review by fmvilas — DISMISSED** [review]

**Comment by derberg** [comment]
@Souvikns just curious why we need to ignore so many files in the case of Sonar Cloud scans?

**Comment by Souvikns** [comment]
It was showing some function is bigger than it should be, but in this case, it is needed, and in the command directory there were some warnings for the function naming conventions, which were needed by oclif to work correctly,

**Comment by derberg** [comment]
@Souvikns but we are ignoring not specific file but all `src/help/**` with wildcard.

- so for the cognitive complexity instead of ignoring in config I suggest you just put `/* eslint-disable sonarjs/cognitive-complexity */` before the complex function that can't be simpler
- in case of naming conflict, just ignore specific files and add comments there why there are ignored so in the future it is clear

**Comment by magicmatatjahu** [comment]
@Souvikns Also, if you need to disable sonar (not eslint, but exactly sonar cloud) only for certain function, you can add comment `// NOSONAR` next to function declaration/type etc:

```
function fn() { // NOSONAR
```

**Comment by Souvikns** [comment]
I will be removing the `sonar.exclusion` property and see what errors I am getting and then I will solve them one by one

**Comment by Souvikns** [comment]
@derberg @magicmatatjahu I have removed all `.sonarcloud.properties` file as there was only exclusion, now sonar cloud is scanning the complete codebase, and I am resolved the code smells.

**Review by derberg — COMMENTED** [review]
@Souvikns there are still linter errors about cognitive complexity. Ignore them the way I wrote in my previous comment. 

we need `"lint": "eslint --max-warnings 5 --config .eslintrc .",` to change to `"lint": "eslint --max-warnings 0 --config .eslintrc .",`

**Comment by Souvikns** [comment]
I have removed the warnings but for some reason, I am getting new code smells, Will be fixing them soon.

**Review by derberg — APPROVED** [review]

**Comment by Souvikns** [comment]
I guess I need @magicmatatjahu and @jotamusik approval to merge

**Comment by derberg** [comment]
@Souvikns at least @magicmatatjahu as he "requested changes" initially and we can't merge before he accepts

**Comment by Souvikns** [comment]
Yeah, I made the changes @magicmatatjahu requested, I have removed the `help` command and now the only `--help` flag works, so now you won't see an extra help command in the root help.

Let me show how the help message is looking right now. 

**Root Help**
image


**Command Help** 
image

**Review by magicmatatjahu — APPROVED** [review]
LGTM! :)

**Comment by Souvikns** [comment]
@jotamusik waiting from your review

**Comment by derberg** [comment]
@Souvikns just fyi you do not need a review of every CODEOWNER, this is needed only when we want to add someone to CODEOWNER file. Imho if you have the approval of 2 maintainers here, you are good to go, unless you synced with Jorge and he needs to have a look.

btw tests are failing, please have a look

**Comment by Souvikns** [comment]
ok @derberg I will merge this after I fix the test, They were working before, but when I updated the branch they are failing.

**Pull request merged**
